### PR TITLE
AMD HIP Support in RAJAPerf

### PR DESCRIPTION
**Pull request by noelchalmers**
This PR adds support for AMD's HIP in the RAJA Performance Suite.

### CODE REVIEW

**Comment by davidbeckingsale** [comment]
@noelchalmers - does this need anything apart from merging the HIP support into RAJA?

**Comment by noelchalmers** [comment]
Nope, this PR should be good to go once the RAJA PR is merged.

**Comment by davidbeckingsale** [comment]
@noelchalmers - this PR needs an updated version of BLT, and also needs to add a dependency on `hip` to the `RAJA_PERFSUITE_DEPENDS` variable inside CMake.

File: src/basic/TRAP_INT-Hip.cpp [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  for ( i = blockDim.x / 2; i > 0; i /= 2 ) {
+    if ( threadIdx.x < i ) {
+      psumx[ threadIdx.x ] += psumx[ threadIdx.x + i ];
+    }
+     __syncthreads();
+  }
+
+#if 1 // serialized access to shared data;
+  if ( threadIdx.x == 0 ) {
+    RAJA::atomic::atomicAdd<RAJA::atomic::hip_atomic>( sumx, psumx[ 0 ] );
```
Comment (davidbeckingsale) [inline src/basic/TRAP_INT-Hip.cpp]: The `atomic` namespace has been removed.
```suggestion
    RAJA::atomicAdd<RAJA::hip_atomic>( sumx, psumx[ 0 ] );
```

File: src/stream/DOT-Hip.cpp [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    if ( threadIdx.x < i ) {
+      pdot[ threadIdx.x ] += pdot[ threadIdx.x + i ];
+    }
+     __syncthreads();
+  }
+
+#if 1 // serialized access to shared data;
+  if ( threadIdx.x == 0 ) {
+    //atomicAdd(dprod, pdot[ 0 ] );
+    RAJA::atomic::atomicAdd(RAJA::atomic::hip_atomic{}, dprod, pdot[ 0 ] );
```
Comment (davidbeckingsale) [inline src/stream/DOT-Hip.cpp]: Atomic namespace has been removed:

```suggestion
    RAJA::atomicAdd(RAJA::hip_atomic{}, dprod, pdot[ 0 ] );
```

**Comment by rhornung67** [comment]
Hi @noelchalmers. I apologize for taking so long to get to this, but I will start updating this to the last RAJA release (v0.12.1) and put up a new PR when it's ready. Do you have any objections to me pulling this into a branch in the PerfSuite repo before I start that?  Thank you very much for all your work on this.

**Pull request merged**